doc fix azurerm_subnet (#23722)
hashicorp/terraform-provider-azurerm: website/docs/r/subnet.html.markdown

* `name` - (Required) The name of the subnet. Changing this forces a new resource to be created.

> * `resource_group_name` - (Required) The name of the resource group in which the virtual network is located.Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `resource_group_name` - (Required) Specifies the resource group in which the virtual network will be located. Changing this forces a new resource to be created.
```

> * `resource_group_name` - (Required) Specifies the resource group in which the virtual network will be located. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `resource_group_name` - (Required) Specifies the resource group in which the subnet will be located. Changing this forces a new resource to be created.
```

Author: I could see the suggestion is to change back the virtual network to a subnet. The main idea of this PR is to correct the subnet to virtual network because we cannot deploy subnet as an individual resource in a location. Hence i update from virtual network from subnet

Author: Hi @WodansSon , Looks like you closed the PR and issue. Do you see any issues on this PR ?

Reviewer: Hi @jayendranarumugam, sorry for not leaving an explanation of why I `closed` the PR. It is my opinion that the Issue and the PR are unnecessary at this point in time since the `NOTE on Virtual Networks and Subnets`(at the top of the document), `Description` and the `Example Usage` all point that the subnet resource is a child of the vNet resource and that they should both live in the same resource group. That said, I believe the true fix here would be to remove both the `virtual_network_name` and `resource_group_name` fields and replace them with a new field `virtual_network_id`. By doing this we could infer both of the previous values from this single value. However, that would be a breaking change and would have to wait until v4.0 of the provider to be released or be wrapped in `FourPointOhBeta` feature flags in the code. Does that make sense?

Author: Yep make sense now

Reviewer 2: Leaving it as resource_group_name and virtual_network_name indicates this is already in the same subscription.  All other resources that refer to an ID are usually (not verified on all resources) able to go across subscriptions. A better change would be to name it virtual_network_resource_group_name and virtual_network_name.  Typing the whole ID would be a pain and then lead to building interpolation to pull the ID from the existing code.
